File: 0203-Remove Linked List Elements/0203-Remove Linked List Elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
 class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode dummy(-1);
        ListNode* p = &dummy;
        while (head)
        {
            if (head->val != val)
            {
                p->next = head;
                p = p->next;
            }
            head = head->next;
        }
        
        p->next = nullptr;
        
        return dummy.next;
    }
};

File: 0201-0300/0203-Remove Linked List Elements/0203-Remove Linked List Elements.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        p = dummy = ListNode(-1)
        while head:
            if head.val != val:
                p.next = head
                p = p.next
            head = head.next
        p.next = None
        return dummy.next
